web-components/src/components/inputs/new/ImageField/ImageField.tsx

```diff
@@ -0,0 +1,149 @@
+import { forwardRef, ReactNode, useState } from 'react';
+import { Crop } from 'react-image-crop';
+import { Box, ButtonBase, SxProps } from '@mui/material';
+
+import { Theme } from '../../../../theme/muiTheme';
+import OutlinedButton from '../../../buttons/OutlinedButton';
+import { getImageSrc } from '../../../image-crop/canvas-utils';
+import CropImageModal from '../../../modal/CropImageModal';
+import FieldFormControl from '../FieldFormControl/FieldFormControl';
+import {
+  DEFAULT_IMAGE_EXTENSION,
+  EXTENSION_REGEX,
+} from './ImageField.constants';
+import { useImageFieldStyles } from './ImageField.styles';
+
+interface Props {
+  name: string;
+  label?: string;
+  buttonText?: string;
+  initialFileName?: string;
+  disabled?: boolean;
+  optional?: boolean | string;
+  circularCrop?: boolean;
+  fixedCrop?: Partial<Crop>;
+  children: ReactNode;
+  setValue: (value: string) => void;
+  onUpload?: (imageFile: File) => Promise<string>;
+  sx?: {
+    label?: SxProps<Theme>;
+    button?: SxProps<Theme>;
+  };
+}
+
+export const ImageField = forwardRef<HTMLInputElement, Props>(
```

Comment:
not sure it makes sense to have a storybook for this, but if it does, do you think it's worth adding one?

Reply:
i think i answered my own question, you built a storybook component for the higher-level piece that has all these lower-level components wired together, that makes sense, it's like writing e2e tests instead of unit tests 👍